spec/models/component_spec.rb

```diff
@@ -1,47 +1,76 @@
 require 'rails_helper'
 
 RSpec.describe Component, type: :model do
+  include StubHubConfigApiSupport
   context '#to_service_metadata' do
+
     before(:each) do
       SpComponent.destroy_all
       MsaComponent.destroy_all
-      msa_component.services << msa_service
-      sp_component.services << sp_service
+    end
+
+    def hub_response_for_encryption(entity_id, certificate_value)
+      {
+        issuerId: entity_id,
+        certificate: certificate_value,
+        keyUse: 'Encryption',
+        federationEntityType: 'RP',
+      }.to_json
+    end
+
+    def hub_response_for_signing(entity_id, certificate_value)
+      [{
+        issuerId: entity_id,
+        certificate: certificate_value,
+        keyUse: 'Signing',
+        federationEntityType: 'RP',
+      }].to_json
     end
 
     let(:published_at) { Time.now }
-    let(:msa_component) { create(:new_msa_component_event).msa_component }
-    let(:sp_component) { create(:new_sp_component_event).sp_component }
+    let(:msa_component) { create(:msa_component) }
+    let(:sp_component) { create(:sp_component) }
     let(:root) { PKI.new }
     let!(:upload_signing_certificate_event_1) do
+
+
       create(:upload_certificate_event,
         usage: CERTIFICATE_USAGE::SIGNING,
         value: root.generate_encoded_cert(expires_in: 6.months),
         component: msa_component
       )
     end
     let!(:upload_signing_certificate_event_2) do
+
       create(:upload_certificate_event,
         usage: CERTIFICATE_USAGE::SIGNING,
         value: root.generate_encoded_cert(expires_in: 6.months),
         component: msa_component
       )
     end
     let!(:upload_signing_certificate_event_3) do
+      stub_signing_certificates_hub_request(environment: sp_component.environment, entity_id: sp_service.entity_id)
+      .to_return(body: hub_response_for_signing(sp_service.entity_id, root.generate_encoded_cert(expires_in: 2.months)))
```

Comment:
```suggestion
```
the `.to_return` line is not needed, it returns a different certificate anyway...

same for all the below